Software/src/Settings.cpp

```diff
@@ -1547,13 +1548,14 @@ void Settings::setMoodLampSpeed(int value)
 	m_this->moodLampSpeedChanged(value);
 }
 
-int Settings::getMoodLampLamp()
+QString Settings::getMoodLampLamp()
 {
 	DEBUG_LOW_LEVEL << Q_FUNC_INFO;
-	return value(Profile::Key::MoodLamp::Lamp).toInt();
+	const QString& val = value(Profile::Key::MoodLamp::Lamp).toString();
+	return val.isEmpty() || val.contains(QRegularExpression("^\\d+$")) ? Profile::MoodLamp::LampDefault : val;
```

Comment:
Please add a comment explaining why numbers are not interpreted as script names. Ideally you'd have a map from numbers to the scripts as a compatibility layer. Breaking things people have set up in the past is confusing and in-transparent.

Reply:
I added a migration